packages/grant-explorer/src/features/discovery/RoundMatchAmountBadge.tsx

```diff
@@ -0,0 +1,41 @@
+import { formatUnits, getAddress, zeroAddress } from "viem";
+import { Badge } from "../common/styles";
+import { useToken } from "wagmi";
+import { getPayoutToken } from "../api/utils";
+import { ChainId } from "common";
+
+type RoundCardStatProps = {
+  matchAmount: string;
+  token: string;
+  chainId: ChainId;
+};
+
+const formatter = new Intl.NumberFormat();
+
+export function RoundMatchAmountBadge({
+  chainId,
+  matchAmount,
+  token,
+}: RoundCardStatProps) {
+  const { data } = useToken({
+    address: getAddress(token),
+    chainId: Number(chainId),
+    enabled: Boolean(token && token !== zeroAddress),
```

Comment:
Here there is a boolean coercion (`token &&`) and then a boolean cast. Seems a case of two wrongs trying to make a right. :)

Can the `token` variable ever be an empty string? If not, then this would be enough:

```
enabled: token !== zeroAddres
```

If yes:

```
enabled: token !== '' && token !== zeroAddress
```

Both would be the least amount of mental parsing/execution for their respective case.

Also, the `token` name has some ambiguity when seen in isolation — is it `{ symbol: string, address: string }` or is it `string`? Only few lines below `nativePayoutToken` is more like the former case, for example.

Naming it `tokenAddress` would save the reader the work of dealing with the ambiguity.

Reply:
Updated to `enabled: token !== zeroAddres` and `tokenAddress`